atlasdb-config/src/main/java/com/palantir/atlasdb/http/AtlasDbErrorDecoder.java

```diff
@@ -43,6 +43,13 @@ public Exception decode(String methodKey, Response response) {
         if (response503ButExceptionIsNotRetryable(response, exception)) {
             return new RetryableException(exception.getMessage(), exception, parseRetryAfter(response));
         }
+        if (response429ButExceptionIsNotRetryable(response, exception)) {
+            // We want to retry on the same node (ExceptionRetryBehaviour.RETRY_ON_SAME_NODE) every time we receive
+            // a response with status 429. Therefore, we set retryAfter to null to comply with
+            // ExceptionRetryBehaviour.getRetryBehaviourForException logic.
+            return new RetryableException(exception.getMessage(), exception, null);
```

Comment:
With the current implementation, we guarantee that at least we will retry when receiving a 429. Agreed that the retry logic needs to be rethought in in a follow up PR.